### PR TITLE
flow: Add tests for optional flow reuse during low memory

### PR DESCRIPTION
**Pull request by coledishington**
Add test for verifying that if the memcap is reached, alive flows will be reused with flow.force-reuse on and not be re-used with flow.force-reuse off.

Feature: #6293

Redmine ticket:
https://redmine.openinfosecfoundation.org/issues/6293

### CODE REVIEW

**Comment by coledishington** [comment]
Suricata PR is https://github.com/OISF/suricata/pull/10232

File: tests/flow-force-reuse/flow-force-reuse-off/test.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+args:
+- --set flow.memcap-policy=drop-flow
+- --set flow.prealloc=0
+- --set flow.memcap=3kb
+- --set flow.hash-size=32
+- --set flow.force-reuse=false
+
+checks:
+  - shell:
+      args: grep 'flow.memcap ' stats.log | wc -l
```
Comment (victorjulien) [inline tests/flow-force-reuse/flow-force-reuse-off/test.yaml:12]: prefer eve.json stats checks here

File: tests/flow-force-reuse/flow-force-reuse-on/test.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+args:
+- --set flow.memcap-policy=drop-flow
+- --set flow.prealloc=0
+- --set flow.memcap=3kb
+- --set flow.hash-size=32
+- --set flow.force-reuse=true
+
+checks:
+  - shell:
+      args: grep -c 'flow[.]get_used ' stats.log
```
Comment (victorjulien) [inline tests/flow-force-reuse/flow-force-reuse-on/test.yaml:12]: eve.json

**Review by victorjulien — CHANGES_REQUESTED** [review]
see inline

File: tests/flow-force-reuse/flow-force-reuse-off/test.yaml (hunk 1 above)
Comment (coledishington) [reply tests/flow-force-reuse/flow-force-reuse-off/test.yaml:12]: The number of counters on flow.memcap and flow.get_used isn't always the same, so the grep is just checking if there is any. Is this comment to use the match syntax of test.yaml or continue using grep but grep eve.json instead?

Thanks

**Comment by coledishington** [comment]
Updated for PR in https://github.com/OISF/suricata-verify/pull/1614

**Pull request closed without merging**